bundles/binding/org.openhab.binding.caldav-command/src/main/java/org/openhab/binding/caldav_command/internal/CalDavBinding.java

```diff
@@ -0,0 +1,509 @@
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.openhab.binding.caldav_command.CalDavBindingProvider;
+
+import org.openhab.core.binding.AbstractBinding;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TypeParser;
+import org.openhab.io.caldav.CalDavEvent;
+import org.openhab.io.caldav.CalDavLoader;
+import org.openhab.io.caldav.EventNotifier;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+
+/**
+ * This is the default implementation of the caldav binding
+ * 
+ * 
+ * BEGIN:ItemName:ON
+ * BEGIN:ItemName:23
+ * 
+ * END:ItemName:OFF
+ * END:ItemName:16
+ * 
+ * @author Robert Delbrück
+ * @since 1.7.0
+ */
+public class CalDavBinding extends AbstractBinding<CalDavBindingProvider> implements ManagedService, EventNotifier {
+	private static final DateTimeFormatter FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss");
+	private static final String SCOPE_END = "END";
+	private static final String SCOPE_BEGIN = "BEGIN";
+	
+	private static final String KEY_READ_CALENDARS = "readCalendars";
+
+	private static final Logger logger = 
+		LoggerFactory.getLogger(CalDavBinding.class);
+	
+	private ItemRegistry itemRegistry;
+	
+	private CalDavLoader calDavLoader;
+	
+	private List<String> readCalendars = new ArrayList<String>();
+	
+	// key=item config, value=list of item changes to listen to
+	private ConcurrentHashMap<CalDavNextEventConfig, List<NextEventContainer>> itemNextEventMap = new ConcurrentHashMap<CalDavNextEventConfig, List<NextEventContainer>>();
+	
+	private List<String> disabledItems = new ArrayList<String>();
+	
+	public CalDavBinding() {
+	}
+	
+	public void setItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = itemRegistry;
+	}
+	
+	public void unsetItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = null;
+	}
+	
+	public void setCalDavLoader(CalDavLoader calDavLoader) {
+		this.calDavLoader = calDavLoader;
+		this.calDavLoader.addListener(this);
+	}
+	
+	public void unsetCalDavLoader() {
+		this.calDavLoader.removeListener(this);
+		this.calDavLoader = null;
+	}
+	
+	public void activate() {
+		
+	}
+	
+	public void deactivate() {
+		// deallocate resources here that are no longer needed and 
+		// should be reset when activating this binding again
+	}
+
+	
+	@Override
+	public void updated(Dictionary<String, ?> properties)
+			throws ConfigurationException {
+		if (properties == null) {
+			logger.warn("no configuration found");
```

Comment:
afaik updated() is called in every class which implements ManagedService without taking the service.pid into account. So this warn statements will appear quite offen in the logs and should be removed though.